### ISSUE TITLE
file descriptor leak

### ISSUE THREAD

**Issue by jsferrei** [spotify/folsom#103]
Version: 1.1.1, 1.2.0
We are noticing that AsciiMemcacheClient/ReconnectingClient is leaking file descriptors when a  memcached server is restarted (or stopped).
Can be reproduced with the following client config:
```
MemcacheClientBuilder
          .newStringClient()
          .withAddress("localhost")
          .connectAscii();
```
When memcached is restarted, the ReconnectingClient parent class will usually make about 5 or 6 attempts to reconnect to the server. In our case, we're seeing the fd count for the process grow by 24 per reconnect attempt (or fail?). If you stop memcached completely, this will continue to grow.

**Comment by spkrka** [comment]
I can confirm this bug. I added a test and a potential fix here: #104 

As a workaround for now, you can call `MemcacheClientBuilder.withEventLoopGroup()` and pass in your own static event-loop-group to avoid the leak.

**Comment by spkrka** [comment]
Fixed in release 1.2.1 - should be available soon.
